File: webapp/src/Websocket/types/stairsHeating.ts
export type StairsHeatingType =
    'enableDisable' |
    'get' |
    'increaseTargetTemp' |
    'decreaseTargetTemp' |
    'increaseOutsideLowerTemp' |
    'decreaseOutsideLowerTemp' |
    'increaseOutsideUpperTemp' |
    'decreaseOutsideUpperTemp';


export type StairsHeatingRequest = {
    type: StairsHeatingType;
}

export type StairsHeatingResponse = {
    data?: StairsHeatingData;
    settings: StairsHeatingSettings;
}

export type StairsHeatingData = {
    currentState: boolean;
    temperature: number;
    current: number;
    createdAt: string;
}

export type StairsHeatingSettings = {
    outsideTemperatureRangeFrom: number;
    outsideTemperatureRangeTo: number;
    targetTemperature: number;
    enabled: boolean;
}

File: src/main/kotlin/com/dehnes/smarthome/api/dtos/StairsHeating.kt
package com.dehnes.smarthome.api.dtos

import com.dehnes.smarthome.zwave.StairsHeatingData

enum class StairsHeatingType {
    enableDisable,
    get,
    increaseTargetTemp,
    decreaseTargetTemp,
    increaseOutsideLowerTemp,
    decreaseOutsideLowerTemp,
    increaseOutsideUpperTemp,
    decreaseOutsideUpperTemp,
}
data class StairsHeatingRequest(
    val type: StairsHeatingType,
)
data class StairsHeatingResponse(
    val currentData: StairsHeatingData
)


